### PR TITLE
Add more programs

### PR DESCRIPTION
**Pull request by m2Giles**
Adds:
Chrome, Edge, Emacs, and KeepassXC.

Modifies:
 Incus to include necessary dependencies for incus managed VMs. (WIP)

### CODE REVIEW

**Comment by travier** [comment]
Thanks for the PR! Let's make one PR per app to keep things simpler to review.

- They have a [Flatpak for 1Password](https://support.1password.com/install-linux/#flatpak). Any specific reason to package it in a sysext?
- For dnf weak deps, we can not set that by default until we audit all our existing systexts for potential missing dependencies so this should be an option for now. I have a draft to add a justfile variable for passing options to dnf.
- The /opt bits looks weird. Could we move things properly in `/usr`?
- Can you add the Containerfiles as well for each app? If you would prefer not to then it's fine, I'll add them afterwards.

**Comment by travier** [comment]
Pushed weak deps option in https://github.com/travier/fedora-sysexts/pull/25.

> For 1password, it's to get access to the ssh-integration, expected browser integration, and access to the CLI. Similar reasons exist as well for keepassxc.

👍🏻
 
> I still think we should explicitly say dnf5 to make it easier for other rpm-ostree systems to be used.

Could you clarify why using dnf5 would make things easier here? Is the idea to add support for dnf5 on Fedora 40 Atomic Desktops?

> Yes, I can move everything into /usr. I will check if /opt still gets locked even if nothing is there.

It might mean "re-packaging" things / fixing the packages. Ideally we would not touch /opt & /usr at the time to keep things simpler but we can definitely see that on a case by case basis depending on how complex that makes things.

**Comment by m2Giles** [comment]
> Could you clarify why using dnf5 would make things easier here? Is the idea to add support for dnf5 on Fedora 40 Atomic Desktops?

Yes, this would simplify things on Fedora 40. It also helps if cliwrap was enabled since /usr/bin/dnf then ends up being a wrapper to rpm-ostree. We are looking at unwinding cliwrap down in ublue, but dnf5 still has some unexpected behaviour compared to rpm-ostree with exit codes.

**Comment by m2Giles** [comment]
These were split into the following PR's

#26 Google-Chrome
#27 Microsoft-Edge
#28 Emacs
#29 KeepassXC
#30 1Password (CLI,Gui separated)

Everything is out of `/opt`. This may have some unexpected behavior for 1Password's Gui Client.

File: sysext.just [hunk 1]
```diff
@@ -304,14 +304,14 @@ reset-selinux-labels target:
 
     filecontexts="/etc/selinux/targeted/contexts/files/file_contexts"
     echo "🏷️ Resetting SELinux labels"
-    podman run --rm -ti \
+    ${SUDO} podman run --rm -ti \
```
Comment (travier) [inline sysext.just:307]: Why is this needed?

File: sysext.just [hunk 2]
```diff
@@ -326,12 +326,9 @@ build-erofs target:
     fi
 
     echo "🔒 Creating EROFS sysext ({{compression}})"
-    version=$(podman run --rm -ti \
-        --security-opt label=disable \
-        "{{target}}" \
-        bash -c 'source /etc/os-release ; echo -n ${OSTREE_VERSION}')
+    version=$(podman inspect "{{ target }}" --format '{{' {{ index .Config.Labels "org.opencontainers.image.version" }} '}}' | grep -Po '\K[0-9].*[0-9]')
     arch=$(uname -m | sed 's/_/-/g')
-    ${SUDO} mkfs.erofs -z{{compression}} {{name}}-${version}-${arch}.raw rootfs > /dev/null
```
Comment (travier) [inline sysext.just:334]: Why is this removed?

**Review by travier — COMMENTED** [review]
Commenting here as this common to all your PRs

File: sysext.just (hunk 1 above)
Comment (m2Giles) [reply sysext.just:307]: rootfs is owned by root after the RPM extraction. So need to use a rootful container to relabel.

File: sysext.just (hunk 2 above)
Comment (travier) [reply sysext.just:334]: Hum, but sometimes some files are not readable by the user as they are `root:root` and not "others" readable. polkit rules are a common case.

**Comment by travier** [comment]
I think I understand the confusion now: when running locally, I'm running all of this from a toolbox with podman commands redirected to run rootless from the host so the file permissions match and appear as root in the toolbox.

I have not tested yet running from outside a toolbox, but that would indeed require full root permissions.

**Comment by travier** [comment]
Realized that I should write a small note about the CI here. To avoid spawning a ton of jobs, I manually regenerate the workflow using templates with the `update_workflows.sh` script. It only acts on commited content so the new sysexts have to be commited first, then the script run, then the commit updated to include the workflows.

File: sysext.just (hunk 2 above)
Comment (m2Giles) [reply sysext.just:334]: Yeah, this can be added back in that case. Instead of relying on that assumption.

**Comment by m2Giles** [comment]
I purposefully haven't touched the workflows seeing how you generate them (which I will steal in the future for templated build ideas)

I will add back the SUDO for the mkfs.erofs step.

For the relabel step I'm unsure how you want to handle that since I believe adding the SUDO might break your toolbox setup.

**Comment by travier** [comment]
One option would be to run everything from a container all the times, so that we would never have to use sudo at all.

**Comment by m2Giles** [comment]
I think for the time being I'll remove my changes to sysext.just

**Comment by travier** [comment]
Yeah, sorry, it would be easier if you could keep those changes locally for now until we fix that more comprehensively.

**Comment by m2Giles** [comment]
For the items that attempt to install into /opt, a working solution was to first remove the symlink for /opt. 

Emacs unfortunately has transitive dependency on kernel headers. This seems more like a packaging issue for gcc to require glibc-devel/kernel-headers.

**Comment by travier** [comment]
Merged all PRs. Thanks!
